### PR TITLE
Skip deletion of ako packageInstall when cluster is deleted since AKO…

### PR DESCRIPTION
**Pull request by lubronzhan**
… needs to clean up external resources.

AKO pod needs to clean up resource when cluster is getting deleted. But currently, its packageInstall just got deleted when cluster is deleted so it can't finish clean up. This will block cluster deletion since AKO has finalizer on the cluster

Added a new annotations on ClusterBootstrap
`run.tanzu.vmware.com/skip-packageinstall-deletion`
`SkipDeletePackageInstallAnnotation` is the annotation that contains a list of comma separated packageInstalls to skip when cluster is deleted
for example
```
"run.tanzu.vmware.com/skip-packageinstall-deletion": "cpi,cni,load-balancer-and-ingress-service"
```

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note

```

### CODE REVIEW

File: addons/pkg/util/addon_util.go [hunk 1]
```diff
@@ -87,6 +87,14 @@ func GetPackageMetadata(ctx context.Context, c client.Client, carvelPkgName, car
 	return pkg.Spec.RefName, pkg.Spec.Version, nil
 }
 
+func GetPackageAnnotations(ctx context.Context, c client.Client, carvelPkgName, carvelPkgNamespace string) (map[string]string, error) {
```
Comment (wjun) [inline addons/pkg/util/addon_util.go:90]: Where is this func used?

**Comment by wjun** [comment]
If we skip the packageinstall deletion, when/how do we delete it?

**Comment by lubronzhan** [comment]
We don't delete the packageIntall itself, it will be deleted as part of cluster. Like other packageInstall, cpi/csi/cni

File: addons/pkg/util/addon_util.go (hunk 1 above)
Comment (lubronzhan) [reply addons/pkg/util/addon_util.go]: Good catch

**Review by wjun — APPROVED** [review]
LGTM.

**Review by 12345lcr — APPROVED** [review]

**Comment by 12345lcr** [comment]
LGTM

**Pull request merged**